### PR TITLE
Rename `ModelMetadata.IsCollectionType` and add "real" `ModelMetadataIsCollectionType`

### PR DESCRIPTION
**Pull request by dougbu**
- existing `IsCollectionType` -> `IsEnumerableType`
- use new `IsCollectionType` in a few places

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/ModelMetadata.cs [hunk 1]
```diff
@@ -322,13 +322,30 @@ public bool IsNullableValueType
...
         /// <remarks>
-        /// A collection type is defined as a <see cref="Type"/> which is assignable to
-        /// <see cref="IEnumerable"/>, and is not a <see cref="string"/>.
+        /// A collection type is defined as a <see cref="Type"/> which is assignable to <see cref="ICollection{T}"/>.
         /// </remarks>
         public bool IsCollectionType
         {
             get
             {
+                // Ignore ICollection type. Would require an additional check and interface is not used in MVC.
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/ModelMetadata.cs]: Ignore non-generic ICollection type?

**Comment by rynowak** [comment]
:shipit:

**Comment by dougbu** [comment]
d8d0a1a

**Pull request closed without merging**